app/src/main/kotlin/com/wire/android/ui/common/visbility/VisibilityState.kt

```diff
@@ -8,36 +8,40 @@ import androidx.compose.runtime.saveable.rememberSaveable
 import androidx.compose.runtime.setValue
 
 @Composable
-fun rememberVisibilityState(): VisibilityState {
+fun <Saveable: Any> rememberVisibilityState(saveable: Saveable? = null): VisibilityState<Saveable> {
     val searchBarState = rememberSaveable(
-        saver = VisibilityState.saver()
+        saver = VisibilityState.saver(saveable)
     ) {
         VisibilityState()
     }
 
     return searchBarState
 }
 
-class VisibilityState(isVisible: Boolean = false) {
+class VisibilityState<Saveable: Any>(isVisible: Boolean = false, saveable: Saveable? = null) {
 
     var isVisible by mutableStateOf(isVisible)
         private set
 
+    var savedState by mutableStateOf<Saveable?>(saveable)
```

Comment:
So we are basically declaring `savedState` as Any??

Reply:
Yup, thanks to that we can pass any state when composable is visible